compiler/rustc_codegen_cranelift/src/constant.rs

```diff
@@ -242,9 +243,9 @@ pub(crate) fn codegen_const_value<'tcx>(
 
 fn pointer_for_allocation<'tcx>(
     fx: &mut FunctionCx<'_, '_, 'tcx>,
-    alloc: ConstAllocation<'tcx>,
+    alloc_id: AllocId,
```

Comment:
@bjorn3 I changed this to take an `AllocId` since otherwise the `ConstValue::Indirect` starts with an AllocId, then looks up the `ConstAllocation` for that, and then `pointer_for_allocation` generates a fresh `AllocId` for that   `ConstAllocation` which seems silly.

Reply:
Makes sense.